diff -up SOAP-Lite-0.710.10/lib/SOAP/Transport/HTTP.pm.fix SOAP-Lite-0.710.10/lib/SOAP/Transport/HTTP.pm
--- SOAP-Lite-0.710.10/lib/SOAP/Transport/HTTP.pm.fix	2012-08-03 13:52:58.000000000 +0200
+++ SOAP-Lite-0.710.10/lib/SOAP/Transport/HTTP.pm	2012-08-03 13:54:01.000000000 +0200
@@ -556,9 +556,16 @@ sub handle {
         if ( !$chunked ) {
             my $buffer;
             binmode(STDIN);
-            while ( sysread( STDIN, $buffer, $length ) ) {
-                $content .= $buffer;
-                last if ( length($content) >= $length );
+            if ( defined $ENV{'MOD_PERL'} ) {
+                while ( read( STDIN, $buffer, $length ) ) {
+                    $content .= $buffer;
+                    last if ( length($content) >= $length );
+                }
+            } else {
+                while ( sysread( STDIN, $buffer, $length ) ) {
+                    $content .= $buffer;
+                    last if ( length($content) >= $length );
+                }
             }
         }
 
